pkg/health/condition/check_backup_ready.go

```diff
@@ -48,6 +50,34 @@ func (a *backupReadyCheck) Check(ctx context.Context, etcd druidv1alpha1.Etcd) R
 		message: "Cannot determine etcd backup status",
 	}
 
+	// Special case of etcd not being configured to take snapshots
+	// Do not add the BackupReady condition if backup is not configured
+	if etcd.Spec.Backup.Store == nil {
```

Comment:
Can you also add a case when `etcd.Spec.Backup.Store.Provider` is passed as empty string.

Reply:
Added